arch/arm64/src/bcm2711/bcm2711_fb.c

```diff
@@ -0,0 +1,471 @@
+                                struct fb_videoinfo_s *vinfo)
+{
+  int err;
+  uint32_t bpp;
+  uint32_t xres;
+  uint32_t yres;
+
+  DEBUGASSERT(vtable != NULL);
+  DEBUGASSERT(vinfo != NULL);
+
+  vinfo->nplanes = 1; /* Only one plane supported */
+
+  err = bcm2711_mbox_getdisp(&xres, &yres);
+  if (err < 0)
+    {
+      gerr("Couldn't get display dimensions: %d", err);
+      return err;
+    }
+
+  vinfo->xres = xres;
+  vinfo->yres = yres;
+
+  err = bcm2711_mbox_getdepth(&bpp);
+  if (err < 0)
+    {
+      gerr("Couldn't get display depth: %d", err);
+      return err;
+    }
+
+  /* NOTE: there appears to be no FB_FMT_* descriptors for BGR pixel
+   * ordering, so I will assume RGB always.
+   *
+   * TODO: not even sure how to check for some of the formats. No idea if 16
+   * bpp is the 555 or 565 version.
+   * Should I also be making an attempt to set pixel order to RGB if the
+   * mailbox reports BGR?
+   */
+
+  err = bcm2711_mbox_getalpha(&xres);
+  if (err < 0)
+    {
+      gerr("Couldn't get alpha mode: %d", err);
+      return err;
+    }
+
+  /* TODO: this method ignores the case when the alpha channel is reversed.
+   * Not sure how the frame buffer upper-half can handle that.
+   */
+
+  if (xres == 1)
+    {
+      gwarn("Alpha channel reversed, this is not handled");
+    }
+  else
+    {
+      ginfo("Alpha channel: %s", xres == 0 ? "enabled" : "ignored");
+    }
+
+  switch (bpp)
+    {
+    case 4:
+      DEBUGASSERT(xres == 2);
+      vinfo->fmt = FB_FMT_RGB4;
+      break;
+    case 8:
+      DEBUGASSERT(xres == 2);
+      vinfo->fmt = FB_FMT_RGB8;
+      break;
+    case 16:
+      vinfo->fmt = xres == 0 ? FB_FMT_RGBA16 : FB_FMT_RGB16_555;
+      break;
+    case 24:
+      DEBUGASSERT(xres == 2);
+      vinfo->fmt = FB_FMT_RGB24;
+      break;
+    case 32:
+      vinfo->fmt = xres == 0 ? FB_FMT_RGBA32 : FB_FMT_RGB32;
+      break;
+    default:
+      gerr("Unknown depth of %u bpp", bpp);
+      return -EIO;
+    }
+
+#ifdef CONFIG_FB_OVERLAY
+  vinfo->noverlays = 0; /* No overlays supported */
+#endif
+
+  return 0;
+}
+
+/****************************************************************************
+ * Name: pinephone_getplaneinfo
+ *
+ * Description:
+ *   Get the planeinfo for the framebuffer. (ioctl Entrypoint:
+ *   FBIOGET_PLANEINFO)
+ *
+ * Input Parameters:
+ *   vtable - Framebuffer driver object
+ *   pinfo  - Returned planeinfo object
+ *
+ * Returned Value:
+ *   Zero (OK) on success; a negated errno value is returned on any failure.
+ *
+ ****************************************************************************/
+
+static int bcm2711_getplaneinfo(struct fb_vtable_s *vtable, int planeno,
+                                struct fb_planeinfo_s *pinfo)
+{
+  int err;
+  uint32_t result;
+  struct bcm2711_fb_s *priv;
+
+  DEBUGASSERT(vtable != NULL);
+  DEBUGASSERT(pinfo != NULL);
+  DEBUGASSERT(planeno == 0); /* Only one supported plane */
+
+  priv = (struct bcm2711_fb_s *)vtable;
```

Comment:
suggestion: 
  struct bcm2711_fb_s *priv = (struct bcm2711_fb_s *)vtable;

  DEBUGASSERT(priv != NULL);

This way you don't need to debug assert vtable and will keep priv declaration/attribution in a single line